### PR TITLE
fix: switch to new theme color variables

### LINKED ISSUE #4127: Update color variable usage to v2 color palette
Ensure that all color variables used across Yari are those as defined in https://github.com/mdn/mdn-minimalist/blob/c57d19a3fe55c18c31f0f5887831a1d8ffe1f678/sass/vars/_mdn-web-docs-palette.scss

This step will allow us to easily load various color palettes inside Yari such as the new redesigned color scheme, high contrast mode, etc.

### LINKED ISSUE #4201: Yari released a monster! 😱
See on https://developer.mozilla.org/en-US/docs/MDN/Guidelines/Writing_style_guide where there is a green monster at the top right corner! 😱 😱 😱

It is likely from the class "example-good" and "example-bad" that, if used with \<pre> elements, show these smileys: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Errors/Bad_radix#examples. The CSS shouldn't use these smileys when used on with other elements such as \<p> or \<div>. Seems like some CSS selector changed recently maybe?

### LINKED ISSUE #4202: Yari released a monster! 😱
See on https://developer.mozilla.org/en-US/docs/MDN/Guidelines/Writing_style_guide where there is a green monster at the top right corner! 😱 😱 😱

It is likely from the class "example-good" and "example-bad" that, if used with \<pre> elements, show these smileys: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Errors/Bad_radix#examples. The CSS shouldn't use these smileys when used on with other elements such as \<p> or \<div>. Seems like some CSS selector changed recently maybe?

### PR DESCRIPTION
**Pull request by schalkneethling**
This now used a new base color palette for MDN Web Docs. More importantly, it uses a set of variables that can easily be overridden to enable other themes such as the new color palette for the redesign, high contrast, etc.

This also cleans up Storybook.

Used the list of pages here to test:
https://github.com/mdn/mdn-minimalist/issues/154

fix #4127
fix #4202 

Bumps [@mdn/minimalist](https://github.com/mdn/mdn-minimalist) from 1.0.1 to 2.0.0.
- [Release notes](https://github.com/mdn/mdn-minimalist/releases)
- [Changelog](https://github.com/mdn/mdn-minimalist/blob/main/CHANGELOG.md)
- [Commits](https://github.com/mdn/mdn-minimalist/compare/v1.0.1...v2.0.0)

---
updated-dependencies:
- dependency-name: "@mdn/minimalist"
  dependency-type: direct:development
  update-type: version-update:semver-major
...

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

### CODE REVIEW

**Comment by peterbe** [comment]
For the record, I kicked off [a build](https://github.com/mdn/yari/actions/runs/1031205432) so in about an hour from now we should have https://4127.content.dev.mdn.mozit.cloud/en-US/docs/Web to click around on.

**Comment by peterbe** [comment]
> This also cleans up Storybook.

There's still lots left of Storybook. Is that a mistake? For example, `"@storybook/addon-actions"` is removed but lots of others are still there.

**Comment by schalkneethling** [comment]
> > This also cleans up Storybook.
> 
> There's still lots left of Storybook. Is that a mistake? For example, `"@storybook/addon-actions"` is removed but lots of others are still there.

Nope, that is all good. I cleaned up so we have a clean slate to start building the new components as part of Plus and the redesign.

**Comment by schalkneethling** [comment]
> For the record, I kicked off [a build](https://github.com/mdn/yari/actions/runs/1031205432) so in about an hour from now we should have https://4127.content.dev.mdn.mozit.cloud/en-US/docs/Web to click around on.

Looks like the build is still running. Does it normally take this long?

https://github.com/mdn/yari/actions/runs/1031205432

**Comment by schalkneethling** [comment]
Also @peterbe, the change to the homepage is intentional. As we are going through a transition phase here, I reckon it is OK.

**Review by peterbe — COMMENTED** [review]
> > For the record, I kicked off [a build](https://github.com/mdn/yari/actions/runs/1031205432) so in about an hour from now we should have https://4127.content.dev.mdn.mozit.cloud/en-US/docs/Web to click around on.
> 
> Looks like the build is still running. Does it normally take this long?
> 
> https://github.com/mdn/yari/actions/runs/1031205432

Takes about 1.5h. :(

**Review by peterbe — COMMENTED** [review]
I'm not a big fan of that "green". It's too "strong" or too "bright". Doesn't feel like it's saying "Yes!" but rather "Take note!!".

**Comment by peterbe** [comment]
Oops. Forgot screenshot for my last comment
Screen Shot 2021-07-14 at 3 46 43 PM

PS. I'm *not* going to block the PR just because (and my wife no less) think that green is unpleasant. There might be a really good reason for it that's outside my skillset.

**Review by peterbe — APPROVED** [review]
I guess it works as prescribed. Most of the many small differences just that; small differences. 

The background color green isn't my job to block on in any way but because BCD is a huge part of MDN's identity so it would be great to get someone like @ddbeck to take a look at that too. 
Perhaps other folks from OWD might want to chip in their 2cents on the new green for "good example".

**Comment by schalkneethling** [comment]
Thank you @peterbe - We can totally improve on those but, I do not want to get too stuck on that as it is going to change pretty soon. I am happy to update the green though. Will make an update to the default color palette and get that in.

**Comment by schalkneethling** [comment]
> I'm not a big fan of that "green". It's too "strong" or too "bright". Doesn't feel like it's saying "Yes!" but rather "Take note!!".

https://github.com/mdn/mdn-minimalist/issues/697

**Comment by peterbe** [comment]
> > I'm not a big fan of that "green". It's too "strong" or too "bright". Doesn't feel like it's saying "Yes!" but rather "Take note!!".
> 
> [mdn/mdn-minimalist#697](https://github.com/mdn/mdn-minimalist/issues/697)

But that means that strong green is going to go out in tonight's build.

**Comment by schalkneethling** [comment]
> But that means that strong green is going to go out in tonight's build.

Already fixing that :)

**Comment by schalkneethling** [comment]
> > > I'm not a big fan of that "green". It's too "strong" or too "bright". Doesn't feel like it's saying "Yes!" but rather "Take note!!".
> > 
> > 
> > [mdn/mdn-minimalist#697](https://github.com/mdn/mdn-minimalist/issues/697)
> 
> But that means that strong green is going to go out in tonight's build.

fix incoming - https://github.com/mdn/mdn-minimalist/pull/699

It is a patch release so it will auto-merge.

File: client/src/document/index.scss [hunk 1]
```diff
@@ -44,15 +43,46 @@ a.new {
...
+  .example-bad {
+    &::after {
+      background-color: $mdn-red700;
+      mask-image: url("~@mdn/dinocons/emojis/frown.svg");
+    }
+  }
+
+  .example-good {
+    &::after {
+      background-color: $mdn-green900;
```
[the remark this answers is not in the document]
Comment (tannerdolby) [reply client/src/document/index.scss:72]: I'm a little late to the party on this, but I have to agree. I'm not a huge fan of this highly saturated green color, the old green I much more preferred. If its a contrast ratio deal to go with the "higher saturated" green, I don't think its worth sacrificing visually appearance. I saw this was fixed in in [#698](https://github.com/mdn/mdn-minimalist/pull/698/files)!

File: client/src/document/index.scss [hunk 2]
```diff
@@ -29,7 +28,7 @@
    here for backwards compatibility */
 a.page-not-created,
 a.new {
-  color: $red-300;
+  color: $mdn-red700;
```
[the remark this answers is not in the document]
Comment (tannerdolby) [reply client/src/document/index.scss:31]: This is actually a "regression" as we wen't from a better contrast ratio with the previous:

Contrast Ratio: 3.8:1 (Fails mostly everything except WCAG AA for large text which is 3.5:1)
Foreground: #1b1b1b
Background: #d0454c

Screen Shot 2021-07-27 at 9 58 17 AM

to an even worse contrast ratio with 

Contrast Ratio: 2.1:1 (Fails WCAG AA and AAA)
Foreground: #1b1b1b
Background: #A30008

Screen Shot 2021-07-27 at 9 58 32 AM

**Pull request merged**